extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/HibernateValidatorFactoryCustomizer.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.hibernate.validator;
+
+import org.hibernate.validator.BaseHibernateValidatorConfiguration;
+
+/**
+ * Meant to be implemented by a CDI bean that provides arbitrary customization for the default
+ * {@link javax.validation.ValidatorFactory}.
+ * <p>
+ * All implementations that are registered as CDI beans are taken into account when producing the default
+ * {@link javax.validation.ValidatorFactory}.
+ * <p>
+ * Customizers are applied in the order of {@link javax.annotation.Priority}.
+ */
+public interface HibernateValidatorFactoryCustomizer {
+
+    <T extends BaseHibernateValidatorConfiguration<T>> void customize(T configuration);
```

Comment:
I'm a bit late to the party, but... These generics are not useful at all, as far as I can see you could just do this.

@gsmet Should I send a PR?

```suggestion
    void customize(BaseHibernateValidatorConfiguration<?> configuration);
```